### ISSUE TITLE
Probleme/Bugs mit der TWROR-Berechnung

### ISSUE THREAD

**Issue by Necoro** [portfolio-performance/portfolio#278]
Ich gebe zu, dass ich den TWROR noch nicht en detail verstanden habe. Aber was ich hier anhand eines Beispiels ([XML File](https://data.necoro.eu/test.xml)) nachvollziehen kann, scheint auf einen Bug hinzudeuten (oder zu mindestens Teile des Verhaltens).

Das obige Beispiel enthält nur [ein Wertpapier](https://www.comdirect.de/inf/etfs/detail/uebersicht.html?ID_NOTATION=31122306).
- Im obigen Beispiel wird mir eine Performance von -5% angezeigt. Das allein ist wunderlich, steigen die Kurse doch.
- Setze ich den Kauf vom 07.04. auf den 08.04. wird daraus auf einmal eine Performance von +6%.
- Noch abstruser wird es, wenn ich statt Xetra die Börse Frankfurt auswähle. Diese stellt für den 07.04. keinen Kurs und das wird in der Performance augenscheinlich als 0 gesehen. Ergo wird mit Kauf am 07.04. mir eine Performance von <-50% angezeigt. Setze ich den Kauf auf den 08.04. sind es auf einmal +8%.

Der letzte Punkt scheint mir auf jeden Fall ein Bug, die anderen können auch an meinem Nicht-Verstehen des TWROR liegen.

**Nachtrag**: Die ersten beiden Punkte scheinen sich mit den Kosten erklären zu lassen. Macht ja auch Sinn, die in der Performance mit zu berücksichtigen. Bleibt trotzdem noch der letzte Punkt.

**Comment by buchen** [comment]
Bei den Kursen verhält sich PP so: *Du hast einen Einlieferung vom 23.3. An dem Tag gibt es aber keinen Kurs, darum wird der nächst frühere Kurs als Schlußkurs verwendet. Also der 20. März mit einem Wert von 28,67 Euro. Der Kaufkurs war aber 28,78. Daraus ergibt sich ein kleiner Verlust.* Transaktionskosten mindern die Performance.

In der Tat habe ich Probleme wenn es keine Kurse gibt. Relevant ist immer was unter *Historische Kurse* bei dem Wertpapier zu sehen ist. Wenn da keine Kurse sind, rechne ich mit 0 und die weitere Berechnung geht kaputt. Als mögliche Lösung habe ich immer mal überlegt, zur Not auf die *errechneten* Kurse zurückzugreifen. D.h. wenn es keinen Kurs gibt, den Kurs eine Kauf/Verkauf/Einlieferung/Auslieferung zu nehmen. Wenn es mehrere an einem Tag gibt, dann eben mitteln. Dazu bin nicht aber noch nicht gekommen.

Erklärt das Deinen Punkt 3?

**Comment by Necoro** [comment]
Ja, das erklärt das sehr gut und die Lösung (als nicht Finanzexperte) scheint mir sinnig. Danke auch für den Hinweis mit dem fehlenden Kurs beim ersten Kauf, das war mir gar nicht aufgefallen.

Alternativvorschlag: Wenn dies zusätzliche Kursermitteln aufwändig ist, könnte man dann als Alternative den Nutzer einfach darauf hinweisen, dass es an einem bestimmten Tag keinen Kurs gibt (ich hab das nämlich auch erst durch Zufall festgestellt). Dann kann man im Zweifel ja einfach das ganze einen Tag nach vorne oder nach hinten verlegen, das macht ja in der Regel keinen Unterschied, verhindert aber solche Ausreißer.

Spontan würde mir als Hinweis einfallen:
  - Dialog (könnte aber nervig werden, wenn das häufiger auftritt)
  - in der Übersicht, wo Käufe als grüne Balken markiert sind für solche ohne Börsenkurs den Balken zB rot färben (+ Tooltip)
